src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AcknowledgeClaimCallbackHandler.java

```diff
@@ -76,49 +101,197 @@ private CallbackResponse populateRespondent1Copy(CallbackParams callbackParams)
             .build();
     }
 
+    private CallbackResponse populateRespondentCopyObjects(CallbackParams callbackParams) {
+        var caseData = callbackParams.getCaseData();
+        MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
+        var updatedCaseData = caseData.toBuilder()
+            .respondent1Copy(caseData.getRespondent1());
+
+        if (ofNullable(caseData.getRespondent2()).isPresent()) {
+            updatedCaseData.respondent2Copy(caseData.getRespondent2());
+        }
+        // Show error message if defendant tries to submit response again ONE_V_TWO_TWO_LEGAL_REP
+        if (featureToggleService.isMultipartyEnabled()
+            && (solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORONE)
```

Comment:
Could we extract this logic into a method for each respondent - just to improve readability + understanding